Update info on python support

abadger opened the pull request:
##### SUMMARY
This rolls up ~three~ five docs changes:
* Update documentation to say Python 3 is officially supported
* Adds a few ref targets for some inventory vars
* Add a platform FAQ section
  * The latter includes documentation on how to solve a problem with non-POSIX shell shipped with Solaris 10.
* Correct some :ref: tags in the release_and_maintenance doc
* Fix some ```Explicit markup ends without a blank line;``` errors that were introduced by the plugin template.

##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
devel 2.5
```

abadger committed:
Update the documentation to list Python 3 as official

abadger committed:
Add some reference targets for inventory variables so we can link to docs

abadger committed:
Add a platform FAQ section

Populate it with

* virtualenv info (previously on the python3 support page)
* BSD (Link to the working with BSD page)
* Solaris (Document how to work around the non-POSIX shell on some
  Solaris hosts)

  Fixes #21594

abadger commented:
/cc @acozine

bcoca commented on docs/docsite/rst/installation_guide/intro_installation.rst:
```diff
@@ -202,11 +194,6 @@ To install the newest version, you may need to unmask the ansible package prior
-.. note::
-
-   If you have Python 3 as a default Python slot on your Gentoo nodes (default setting), then you
-   must set ``ansible_python_interpreter = /usr/bin/python2`` in your group or inventory variables.
```
Gentoo defaults to  3.4, so we might want to leave the note but change the content, but probably make it 'some distros' as it is not the only one.

bcoca commented on docs/docsite/rst/reference_appendices/faq.rst:
```diff
@@ -105,26 +105,100 @@ and run Ansible from there.
+  this in inventory like so::
+
+    solaris1 ansible_shell_executable=/usr/xpg4/bin/sh
+
```
+10 on adding solaris notes

abadger committed:
Fix some refs in the release_and_maintenance document

abadger committed:
Fix unindent error in module template

Fix for the module/plugin template unintentionally unindented inside of
a raw block, leading to errors like:

ERROR: docs/docsite/rst/modules/redshift_facts_module.rst:289:0: Explicit markup ends without a blank line; unexpected unindent.

acozine commented on docs/docsite/rst/reference_appendices/faq.rst:
```diff
@@ -105,26 +105,102 @@ and run Ansible from there.
+Running on Solaris
+------------------
+
+Some Solaris machines use a non-POSIX shell by default.  This can cause trouble for things such as
```
How about:
"By default, some Solaris machines run a non-POSIX shell, which does not correctly expand the default tmp directory Ansible uses ( `~/.ansible/tmp`). If you see module failures on Solaris machines, this is likely the problem. There are several workarounds:"

acozine replied:
If this is only on certain modules, then I'd say "If you see failures on the x and/or y modules on Solaris machines"

acozine reviewed (commented):
+1 for the Common Platform section. That addresses issue #21594.

abadger committed:
Make wording for Solaris troubleshooting better.

acozine commented on docs/docsite/rst/reference_appendices/faq.rst:
```diff
@@ -200,6 +200,8 @@ several workarounds:
 
     solaris1 ansible_shell_executable=/usr/xpg4/bin/sh
 
+  (bash, ksh, and zsh should also be POSIX compatible if you have any of those installed).
```
+1 for adding specific examples

acozine commented on docs/docsite/rst/installation_guide/intro_installation.rst:
```diff
@@ -64,40 +61,33 @@ Managed Node Requirements
-   Ansible 2.2 introduces a tech preview of support for Python 3. For more information, see `Python 3 Support <http://docs.ansible.com/ansible/python_3_support.html>`_.
+   * By default, Ansible uses the python interpreter located at :file:`/usr/bin/python` to run its
+     modules.  However, some Linux distributions may only have a Python 3 interpreter installed to
+     :file:`/usr/bin/python3` by default.  On those systems, you can either set the
```
What error or behavior would the user see in this scenario?

abadger replied:
Fixed.  Latest version adds the error to look out for.

acozine reviewed (approved):
WRT issue #21594 this looks good.

abadger commented:
Merged to devel.  Adding to the 2.5.x blocker project for potential inclusion in 2.5.2

abadger commented:
cherry-picked to stable-2.5 for inclusion in 2.5.2
